inc/namespace.php

```diff
@@ -128,6 +132,9 @@ function is_disabled( WP_User $user ) : bool {
  * @return void
  */
 function disable_user( WP_User $user ) : void {
+	// Notify that this user is going to be disabled.
+	do_action( DISABLE_USER_ACTION, $user->ID, $user );
```

Comment:
I'd suggest `pre_disable` or something here to make it explicitly before disabling - and maybe make a filter to allow cancelling?

Reply:
I like the filter idea, I chose this naming convention as that's what core uses for `delete_user`/`deleted_user`

Reply:
Yeah, it's slightly more general use with a filter - I've always hated those core ones because they're an easy footgun.